plugins/apollo-explorer/README.md

```diff
@@ -69,3 +69,41 @@ That's it! You should now see an `Apollo Explorer` item in your sidebar, and if
 Once you authenticate, your graph is ready to use 🚀
 
 ![Logged In](./docs/img/logged-in.png)
+
+### Authentication Tokens for Apollo Studio
+
+If you need to utilize an ApiRef to supply a token to Apollo, you may do so using an ApiHolder.
+
+In `packages/app/src/App.tsx` perform the following modifications from above. The import `ssoAuthApiRef` is used as an example
+
+```typescript
+import { ApolloExplorerPage, EndpointProps } from '@backstage/plugin-apollo-explorer';
+import { ssoAuthApiRef } from '@backstage/devkit';
+import { ApiHolder } from '@backstage/core-plugin-api';
+
+
+async function apolloPluginEndpointsCallback(options: { apiHolder: ApiHolder }): Promise<EndpointProps[]> {
+  const sso = options.apiHolder.get<any>(ssoAuthApiRef)
+  return [{
+    title: 'Github',
+    graphRef: 'my-github-graph-ref@current',
+    initialState: {
+      headers: {
+        authorization: `Bearer ${await sso.getToken()}`,
```

Comment:
This token will eventually expire if you stay on the page for too long. We'll need something that's called on every request rather than only when loading the endpoint list.

What I think we'll want to do instead is to forward the `handleRequest` option, where we could also forward the `apiHolder` in the second options argument of that callback. I think there's also a potential security benefit with that since the `headers` option strikes me as something that might be a bit more visible and prone to leakage than using `handleRequest`.

Reply:
I'm not sure I followed completely, however [84c4560](https://github.com/backstage/backstage/pull/23405/commits/84c4560c4fa9862a9560bbc7c9431875086b8d84) might do it?